### DISCUSSION TITLE
Heat Pump Heater Available Schedule

### DISCUSSION THREAD

**Discussion by StrikerS7R** [tagyoureit/nodejs-poolController#607] [App and O/S Config Questions] [answered]
I have a Raypak 8450ti heat pump heater working away to maintain the pool temp to spouse friendly temps and given the fall weather approaching would like to set a schedule so the heat pump is only enabled to run during the warm parts of the day. Currently it's triggered whenever the pump is running and the pool temp is below the set point. This makes basic sense but isn't as efficient since it's a large heat pump and only needs to run a portion of the time to maintain pool temp and it's obviously way more efficient to run in the warmer afternoon vs. the cool mornings when the VS pump starts the low speed filtering.

I saw there's pump circuits labeled Heat Enable, Heat Boost or Pool Heat Enable but since I'm on a pure Nixie setup I'm not familiar with the Pentair controller logic and haven't found a helpful guide that would educate me on how to set that up since it sounds promising for my goal but don't want to break a functioning setup monkeying around with it.

Is this capability possible with Nixie and if so can you point me to a guide/info on how to set that up?

**Comment by rstrouse** [comment]
Set up your schedules for the pool circuit that enable the heat pump during the warm parts of the day and disables them during the cool parts of the day.  It will change the heat settings for the body when the schedule kicks in.  You can set the heat mode as well as the setpoint.

**Reply by StrikerS7R** [reply]
>Set up your schedules for the pool circuit that enable the heat pump...

How do I define which pool circuit enables the heat pump?

I have the heat pump fireman switch on a relay controlled through REM and connected in the Dashboard to REM > Heater (Relay # 6)
![image](https://user-images.githubusercontent.com/22134823/190234787-f90fb05e-4b75-41ee-97a6-38d31c2eeae7.png)

I also have an Aux-Circuit named "Heater" Type "Generic" to Connection: REM > Device: Heater (Relay # 6)
![image](https://user-images.githubusercontent.com/22134823/190234747-d95307de-062f-42c3-8e62-ae3b09e4931f.png)

I can add more Circuits to the pump but only get 4 relay/program options (presumably since it's a SuperFlo). I suspect this is probably the route but also feels like could very much not be and I'm reading into it what I want to see. I can't use Pool Heat Enable on more than one relay/program since it gives the error "Pump Circuits must be unique" when trying to apply it to the existing programs.
![image](https://user-images.githubusercontent.com/22134823/190235374-4369a82f-98cd-4c02-935a-62cf0ec2b437.png)
![image](https://user-images.githubusercontent.com/22134823/190235441-125d59c0-4f88-44a1-84f5-b395b3889538.png)
![image](https://user-images.githubusercontent.com/22134823/190236045-134fdea3-592d-44ae-994f-49fde9ee9985.png)

**Reply by tagyoureit** [reply]
@rstrouse means on the schedules tab like such:

![2022-09-14_11-46-53](https://user-images.githubusercontent.com/1646840/190237933-39c45d2a-50df-4384-83e7-2ee42d2fe250.png)

Notice the top schedule is 8am-1pm and the lower one is 1pm-5pm as an example.

Don't use the `Heat Enable, Heat Boost or Pool Heat Enable` special circuits.  They are leftover from the OCP spa side remotes aren't wired up to anything in Nixie.

**Comment by rstrouse** [comment, marked as answer]
... @tagyoureit beat me to it.

You do not need or want that heater Aux-Circuit.  Then set up the Schedules under the Schedules tab for the Pool.  If you want a minimum speed then set the Any Heater circuit to be that speed.
![image](https://user-images.githubusercontent.com/47839015/190237967-add48def-f4ac-43a1-b8a1-a58e3a92ea69.png)

**Reply by StrikerS7R** [reply]
Ahhhh!! I didn't get/understand the schedule part of setting a Pool specific Circuit since the Heat Source options wasn't visible on any of the other circuits associated with the pump speeds. 

@tagyoureit Can you elaborate on why you have a Heat Source Off schedule? Wouldn't that be anytime a Heat Source isn't selected?
![image](https://user-images.githubusercontent.com/22134823/190243556-da314b69-0b0b-4123-9399-9e0a4670b8e6.png)

That minimum speed part is perfect and was my next thought, thank you!
![image](https://user-images.githubusercontent.com/22134823/190242917-5de715f6-d7fc-4b64-acb2-88dc708ae044.png)

**Reply by tagyoureit** [reply]
> Can you elaborate on why you have a Heat Source Off schedule? Wouldn't that be anytime a Heat Source isn't selected?

You asked above `would like to set a schedule so the heat pump is only enabled to run during the warm parts of the day ... way more efficient to run in the warmer afternoon vs. the cool mornings`.  That's what the 2 schedules do.  They turn off the heat pump in the morning and then turn on the heat pump in the afternoon.  Think about it like this - when the schedule starts it will go to the heat settings and change the heat source.  Setting it to none in the morning will prevent the heat pump from running.  Setting it to heat pump on the afternoon schedule will then reflect the setpoint that you want it to reach.

**Reply by StrikerS7R** [reply]
I get that part but to be more specific, does the heat source need to be turned off? 

I.e. Which, if either, state is latching? Does it need to be turned off to prevent it running or need to be specifically turned on to allow it to run?

**Reply by rstrouse** [reply]
The Aux Circuit that you defined as Heater should be removed, that is in direct conflict with the heater control.  The schedule when it kicks off will set the heat mode for the Pool body as it is defined on the schedule.  So if the schedule starts at 8am and the Heat Source is set to off it will deselect the heat mode when it starts the pool body.  Then later in the afternoon if you have the Heat Source set to Heat Pump it will select the heat pump into the heat mode and start heating the pool body.

**Reply by tagyoureit** [reply]
EDIT:  Saw your config above.  To clarify, you can remove the aux circuit because you have the relay selected via the heater.  When that is set up that way the pool circuit will turn on the heater when the following conditions are true:
1. Heat mode is heat pump or heat pump preferred
2. The heat set point is above the current pool temp

**Reply by StrikerS7R** [reply]
Copy that, the Heater Aux-Circuit has been removed and it's working happily as you've described using the schedule for the default Pool circuit and heat source selection. Thanks for the clarification
